extensions/amp-a4a/0.1/template-renderer.js

```diff
@@ -15,34 +15,53 @@
  */
 
 import {FriendlyFrameRenderer} from './friendly-frame-renderer';
+import {Renderer} from './amp-ad-type-defs';
 import {getAmpAdTemplateHelper} from './template-validator';
 
 /**
  * Render AMP creative into FriendlyFrame via templatization.
  */
-export class TemplateRenderer extends FriendlyFrameRenderer {
+export class TemplateRenderer extends Renderer {
+
+  /**
+   * Constructs a TemplateRenderer instance.
+   */
+  constructor() {
+    super();
+
+    /**
+     * @type {!FriendlyFrameRenderer}
+     * @protected
+     */
+    this.friendlyFrameRenderer_ = new FriendlyFrameRenderer();
+  }
+
   /** @override */
-  render(context, element, validatorData) {
-    const {creativeData} = validatorData;
-    return super.render(context, element, creativeData).then(() => {
-      const templateHelper = getAmpAdTemplateHelper(context.win);
-      return templateHelper
-          .render(
-              creativeData.templateData.data,
-              this.iframe.contentWindow.document.body)
-          .then(renderedElement => {
-            const {analytics} = creativeData.templateData;
-            if (analytics) {
-              templateHelper.insertAnalytics(renderedElement, analytics);
-            }
-            // This element must exist, or #render() would have thrown.
-            const templateElement = this.iframe.contentWindow.document
-                .getElementsByTagName('template')[0];
-            this.iframe.contentWindow.document.body
-                .removeChild(templateElement);
-            this.iframe.contentWindow.document.body
-                .appendChild(renderedElement);
-	  });
-    });
+  render(context, element, creativeData) {
+    return this.friendlyFrameRenderer_.render(context, element, creativeData)
+        .then(() => {
+          const templateData =
+          /** @type {!./amp-ad-type-defs.AmpTemplateCreativeDef} */ (
+              creativeData.templateData);
+          const {data} = templateData;
+          if (!data) {
+            return Promise.resolve();
+          }
+          const templateHelper = getAmpAdTemplateHelper(context.win);
+          const {iframe} = this.friendlyFrameRenderer_;
```

Comment:
once you made `friendlyFrameRenderer` an util class, then its `render()` can return this `iframe`.

Reply:
Done.